indigo_api/models/works.py

```diff
@@ -312,6 +313,27 @@ def friendly_type(self):
         if plugin:
             return plugin.work_friendly_type(self)
 
+    def possible_expression_dates(self):
+        initial = self.publication_date or self.commencement_date
+        amendment_dates = [a.date for a in self.amendments.all()]
+        consolidation_dates = [c.date for c in self.arbitrary_expression_dates.all()]
+        dates = [
+            {'date': date,
+             'amendment': date in amendment_dates,
+             'consolidation': date in consolidation_dates,
+             'initial': date == initial}
+            for date in amendment_dates + consolidation_dates
+        ]
+
+        if initial not in amendment_dates + consolidation_dates:
+            dates = dates + [
```

Comment:
For style, a common way of formatting dicts that span multiple lines is to put the opening `{` and closing `}` on their own lines. It makes it easier to read and add/remove entries. A trailing comma on the last entry also makes it easier to add/shuffle lines around later without having to tweak commas. eg:

```
dates = dates + [{
    'date': initial,
    'initial': True,
}]
```

Reply:
Also, rather than concatenating two lists, `dates.append({...})` is better.